graphix/linalg_validations.py

```diff
@@ -15,15 +15,15 @@ def is_square(matrix: npt.NDArray[_T]) -> bool:
     if matrix.ndim != 2:
         return False
     rows, cols = matrix.shape
-    return rows == cols
+    return bool(rows == cols)
 
 
 def is_qubitop(matrix: npt.NDArray[_T]) -> bool:
     """Check if matrix is a square matrix with a power of 2 dimension."""
     if not is_square(matrix):
         return False
     size, _ = matrix.shape
-    return size > 0 and size & (size - 1) == 0
+    return bool(size > 0 and size & (size - 1) == 0)
```

Comment:
I still don't understand why newer versions of `numpy` don't provide accurate information about `matrix.shape`, but there may be no better alternative at this time.

Reply:
I just submitted a lighter fix in e35258e. Note that this circumvents a bug that only affects numpy 2.1: the type of the size property is `Any` in numpy 2.1, but in numpy 2.0 and numpy >=2.2 it is a tuple of ints.